gordo/client/io.py

```diff
@@ -16,15 +15,56 @@ class HttpUnprocessableEntity(BaseException):
     pass
 
 
-def _handle_response(resp: requests.Response) -> Union[dict, bytes]:
+class ResourceGone(BaseException):
+    """
+    Represents an error from an HTTP status code of 410: Gone.
+    Indicates that access to the target resource is no longer available at the origin
+    server and that this condition is likely to be permanent.
+
+    Used in our case for calling the server with a revision which is no longer used.
+    """
+
+    pass
+
+
+class BadGordoRequest(BaseException):
+    """
+    Represents a general 4xx bad request
+    """
+
+    pass
+
+
+class NotFound(BaseException):
+    """
+    Represents a 404
+    """
+
+    pass
```

Comment:
Should we be using the werkzeug or requests' exceptions?

Reply:
I avoided werkzeug because I thought it would be nice to be able to seperate out the server as a extras_require, and then it would be cool not to require werkzeug in the client-part